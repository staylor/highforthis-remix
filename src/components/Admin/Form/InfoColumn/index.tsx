import type { Fields } from '@/types';
import cn from 'classnames';
import type { ReactNode } from 'react';

const border = cn('border border-detail');

const InfoBox = ({ children, ...props }: { children: ReactNode }) => (
  <aside className={cn(border, 'mb-5 box-border block shadow')} {...props}>
    {children}
  </aside>
);

const Content = ({ children }: { children: ReactNode }) => (
  <div className="px-5 pt-1.5 pb-5 text-sm">{children}</div>
);

interface InfoColumnProps {
  infoFields: Fields;
  metaFields: Fields;
  label?: string;
  button?: ReactNode;
}

export default function InfoColumn({
  infoFields = [],
  metaFields = [],
  label = '',
  button = null,
}: InfoColumnProps) {
  return (
    <section className="md:-mr-75 md:w-70 sticky top-2.5 mt-2.5 mr-0 w-full md:float-right">
      {infoFields.length > 0 ? (
        <InfoBox>
          <h3 className={cn(border, 'text-dark select-none py-2 px-3 text-sm font-bold')}>
            {label}
          </h3>
          <Content>
            <>{infoFields}</>
            {button}
          </Content>
        </InfoBox>
      ) : null}
      {metaFields.length > 0 ? (
        <InfoBox>
          <Content>
            <>{metaFields}</>
          </Content>
        </InfoBox>
      ) : null}
    </section>
  );
}
