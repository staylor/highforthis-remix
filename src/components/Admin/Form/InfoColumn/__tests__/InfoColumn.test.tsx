import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import InfoColumn from '@/components/Admin/Form/InfoColumn';

const label = 'Info!';

const button = <button type="button">Submit</button>;

const infoFields = [
  <p key="0">
    <input name="foo" />
  </p>,
];
const metaFields = [
  <p key="0">
    <input name="bar" />
  </p>,
];

describe('InfoColumn', () => {
  test('No props', () => {
    const { container } = render(<InfoColumn />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with infoFields', () => {
    const { container } = render(<InfoColumn infoFields={infoFields} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with metaFields', () => {
    const { container } = render(<InfoColumn metaFields={metaFields} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with info and meta fields', () => {
    const { container } = render(<InfoColumn infoFields={infoFields} metaFields={metaFields} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('no button', () => {
    const { container } = render(
      <InfoColumn label={label} infoFields={infoFields} metaFields={metaFields} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with button', () => {
    const { container } = render(
      <InfoColumn label={label} infoFields={infoFields} metaFields={metaFields} button={button} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
