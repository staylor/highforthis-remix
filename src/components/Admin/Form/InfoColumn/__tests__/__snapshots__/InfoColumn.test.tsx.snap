// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoColumn > No props 1`] = `
<section
  class="sticky top-2.5 mr-0 mt-2.5 w-full md:float-right md:-mr-75 md:w-70"
/>
`;

exports[`InfoColumn > no button 1`] = `
<section
  class="sticky top-2.5 mr-0 mt-2.5 w-full md:float-right md:-mr-75 md:w-70"
>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <h3
      class="border border-detail text-dark select-none py-2 px-3 text-sm font-bold"
    >
      Info!
    </h3>
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="foo"
        />
      </p>
    </div>
  </aside>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn > with button 1`] = `
<section
  class="sticky top-2.5 mr-0 mt-2.5 w-full md:float-right md:-mr-75 md:w-70"
>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <h3
      class="border border-detail text-dark select-none py-2 px-3 text-sm font-bold"
    >
      Info!
    </h3>
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="foo"
        />
      </p>
      <button
        type="button"
      >
        Submit
      </button>
    </div>
  </aside>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn > with info and meta fields 1`] = `
<section
  class="sticky top-2.5 mr-0 mt-2.5 w-full md:float-right md:-mr-75 md:w-70"
>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <h3
      class="border border-detail text-dark select-none py-2 px-3 text-sm font-bold"
    />
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="foo"
        />
      </p>
    </div>
  </aside>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn > with infoFields 1`] = `
<section
  class="sticky top-2.5 mr-0 mt-2.5 w-full md:float-right md:-mr-75 md:w-70"
>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <h3
      class="border border-detail text-dark select-none py-2 px-3 text-sm font-bold"
    />
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="foo"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn > with metaFields 1`] = `
<section
  class="sticky top-2.5 mr-0 mt-2.5 w-full md:float-right md:-mr-75 md:w-70"
>
  <aside
    class="border border-detail mb-5 box-border block shadow"
  >
    <div
      class="px-5 pb-5 pt-1.5 text-sm"
    >
      <p>
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;
